lib/account/pages/new_address_page.dart

```diff
@@ -0,0 +1,196 @@
+                        ),
+                      ),
+                      const SizedBox(width: FlexSizes.inputFieldSpacing),
+                      Expanded(
+                        child: FormBuilderTextField(
+                          name: 'postalCode',
+                          decoration: const InputDecoration(
+                              labelText: 'Postal Code/Zip'),
+                          validator: FormBuilderValidators.required(),
+                        ),
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: FlexSizes.inputFieldSpacing),
+
+                  // phone number and mobile number
+                  Row(
+                    children: [
+                      Expanded(
+                        child: FormBuilderTextField(
+                          name: 'phone',
+                          decoration: const InputDecoration(
+                              labelText: 'Phone number (optional)'),
+                        ),
+                      ),
+                      const SizedBox(width: FlexSizes.inputFieldSpacing),
+                      Expanded(
+                        child: FormBuilderTextField(
+                          name: 'cellphone',
+                          decoration: const InputDecoration(
+                              labelText: 'Mobile number (optional)'),
+                        ),
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: FlexSizes.spacerSection),
+
+                  ElevatedButton(
+                    onPressed: () {
+                      if (_formKey.currentState!.saveAndValidate()) {
+                        context.read<AddressCubit>().saveAddress(
+                              countryIsoCode: _formKey
+                                  .currentState!.fields['country']!.value,
+                              firstName: _formKey
+                                  .currentState!.fields['firstName']!.value,
+                              lastName: _formKey
+                                  .currentState!.fields['lastName']!.value,
+                              line1:
+                                  _formKey.currentState!.fields['line1']!.value,
+                              line2:
+                                  _formKey.currentState!.fields['line2']!.value,
+                              phone:
+                                  _formKey.currentState!.fields['phone']!.value,
+                              postalCode: _formKey
+                                  .currentState!.fields['postalCode']!.value,
+                              regionIsoCode: _formKey
+                                  .currentState!.fields['region']!.value,
+                              town:
+                                  _formKey.currentState!.fields['town']!.value,
+                            );
```

Comment:
I think a cleaner way to handle saving data to the backend is to configure the formBuilder to return its JSON in the shape expected by the backend, so no need to build an Address model that is re-transformed into json when sent to the backend.

So we can simply do `context.read<AddressCubit>().saveAddress(_formKey.currentState)` here
No need to adjust the model definition (`id` nullable)
Even no need for the toJson method